### PR TITLE
Fix/firstclass changes

### PR DESCRIPTION
**Pull request by wepee**

### CODE REVIEW

File: contracts/LingoNFT.sol [hunk 1]
```diff
@@ -37,37 +37,31 @@ contract LingoNFT is ERC721, EIP712, Ownable, ReentrancyGuard {
     }
 
     /// @dev Mapping to store Tier for each token ID
```
Comment (au2001) [inline contracts/LingoNFT.sol]: Out of date

File: contracts/LingoNFT.sol [hunk 2]
```diff
@@ -37,37 +37,31 @@ contract LingoNFT is ERC721, EIP712, Ownable, ReentrancyGuard {
...
     struct MintData {
         address sender;
         Tier tier;
     }
 
+    string public firstClassURI = "ipfs://QmWsdztWmpXf8hiuGX8p8sXdn6K6J6zWuWWMcgaa6TaP2H";
+    string public privateJetURI = "ipfs://QmPNSZ2jgQtLnk46EaCvcm5WQ31PZDMeCtgtPfbBbtBMsx";
+
     /// @dev Tracks whether an address has minted for a specific tier
     mapping(address => mapping(Tier => bool)) private _hasMinted;
```
Comment (au2001) [inline contracts/LingoNFT.sol]: Ordering does not follow standard convention
Use solhint

File: contracts/LingoNFT.sol [hunk 3]
```diff
@@ -90,7 +84,7 @@ contract LingoNFT is ERC721, EIP712, Ownable, ReentrancyGuard {
     /// @param uri The new URI
     function setTierURI(Tier tier, string calldata uri) external onlyOwner {
         require(bytes(uri).length > 0, "URI cannot be empty");
-        _tierURIs[tier] = uri;
+        tier == Tier.FIRST_CLASS ? firstClassURI = uri : privateJetURI = uri;
```
Comment (au2001) [inline contracts/LingoNFT.sol]: Use an if-else

File: contracts/LingoNFT.sol (hunk 2 above)
Comment (au2001) [inline contracts/LingoNFT.sol]: A nested mapping is wasteful, use a mapping of structs
(If there were more than 32 tiers, you would have to use OpenZeppelin's lib)

File: contracts/LingoNFT.sol [hunk 4]
```diff
@@ -1,17 +1,17 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.18;
 
-import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
 import "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
 import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
 import "@openzeppelin/contracts/utils/Counters.sol";
```
Comment (au2001) [inline contracts/LingoNFT.sol]: Remove this lib (deprecated)

Comment (au2001) [inline contracts/LingoNFT.sol]: Use named imports

File: contracts/LingoNFT.sol [hunk 5]
```diff
@@ -99,30 +93,16 @@ contract LingoNFT is ERC721, EIP712, Ownable, ReentrancyGuard {
...
         uint8 v
-    ) external nonReentrant isActive {
-        require(
-            tier == Tier.FIRST_CLASS,
-            "This function is only for FIRST_CLASS tier"
-        );
+    ) external isActive {
         require(
             !_hasMinted[msg.sender][Tier.FIRST_CLASS],
             "Address already minted First"
```
Comment (au2001) [inline contracts/LingoNFT.sol]: Don't use deprecated string reverts

File: contracts/LingoNFT.sol [hunk 6]
```diff
@@ -192,27 +161,26 @@ contract LingoNFT is ERC721, EIP712, Ownable, ReentrancyGuard {
...
 
     /// @notice Mints a new token of a specified tier to a given address
     /// @dev Internal function that handles the logic for minting new tokens
     /// @param to The address to mint the token to
     /// @param tier The tier of the new token
     function _mint(address to, Tier tier) internal {
-        require(to != address(0), "Cannot mint to the zero address");
-
         _tokenIdCounter.increment();
         uint256 tokenId = _tokenIdCounter.current();
```
Comment (au2001) [inline contracts/LingoNFT.sol]: Waste of gas

File: test/test.js [hunk 7]
```diff
@@ -16,12 +16,16 @@ describe("Lingo NFT Tests", async () => {
         wrongSigner,
     ] = await ethers.getSigners();
 
-    messageHash = ethers.utils.id("hello");
+    const [FIRST_CLASS_TIER, PRIVATE_JET_TIER] = [0, 1];
```
Comment (au2001) [inline test/test.js]: Use an enum or two separate statements

File: test/test.js [hunk 8]
```diff
@@ -16,12 +16,16 @@ describe("Lingo NFT Tests", async () => {
...
+
+    const [FIRST_CLASS_URI, PRIVATE_JET_URI] = ["ipfs://QmWsdztWmpXf8hiuGX8p8sXdn6K6J6zWuWWMcgaa6TaP2H", "ipfs://QmPNSZ2jgQtLnk46EaCvcm5WQ31PZDMeCtgtPfbBbtBMsx"]
```
Comment (au2001) [inline test/test.js]: Use a Record

File: test/test.js [hunk 9]
```diff
@@ -559,22 +560,36 @@ describe("Lingo NFT Tests", async () => {
...
     });
 
     describe("Airdrop NFT", function () {
         let recipientAddresses = ["0x9965507D1a55bcC2695C58ba16FB37d819B0A4dc", "0x976EA74026E726554dB657fA54763abd0C3a0aa9", "0x14dC79964da2C08b23698B3D3cc7Ca32193d9955", "0x23618e81E3f5cdF7f54C3d65f7FBc0aBf5B21E8f"]
-        let firstClassTier = 0;
-        let privateClassTier = 1;
 
         it("should revert if not called by the owner", async function () {
-            await expect(lingoNFT.connect(user1).airdropNFT(recipientAddresses, firstClassTier, {
+            await expect(lingoNFT.connect(user1).airdropNFT(FIRST_CLASS_TIER,recipientAddresses, {
```
Comment (au2001) [inline test/test.js]: Run prettier

File: contracts/LingoNFT.sol [hunk 10]
```diff
@@ -1,17 +1,17 @@
...
 import "@openzeppelin/contracts/access/Ownable.sol";
 import "@openzeppelin/contracts/utils/Counters.sol";
-import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
 
 /// @title LingoNFT: An NFT Contract for Lingo SocialFi Campaign
 /// @notice This contract allows for the minting and management of various NFT tiers.
 /// @dev Inherits ERC721 for NFT functionality, Ownable for access control, and ReentrancyGuard for security.
-contract LingoNFT is ERC721, EIP712, Ownable, ReentrancyGuard {
+contract LingoNFT is ERC721URIStorage, EIP712, Ownable {
```
Comment (au2001) [inline contracts/LingoNFT.sol]: Why did you add `ERC721URIStorage` if it's not to use it?

**Review by au2001 — CHANGES_REQUESTED** [review]

**Comment by au2001** [comment]
This PR has 2M+ changes and >5k files
If you want a proper review, please clean it up

File: contracts/LingoNFT.sol [hunk 11]
```diff
@@ -1,188 +1,210 @@
...
+
+    /// @notice Airdrops NFTs of a specified tier to multiple addresses
+    /// @param tier The tier of NFTs to airdrop
+    /// @param recipients The addresses to receive the NFTs
+    function airdropNFT(
+        Tier tier,
+        address[] calldata recipients
+    ) external onlyOwner {
+        for (uint256 i = 0; i < recipients.length; i++) {
+            _mint(recipients[i], tier);
```
Comment (au2001) [inline contracts/LingoNFT.sol:145]: This bypasses the `maxFirstClassSupply` check

File: contracts/LingoNFT.sol [hunk 12]
```diff
@@ -1,188 +1,210 @@
...
+error Unauthorized();
 
-    // Mapping to store URI for each token
-    mapping(uint256 => string) private _tokenURIs;
-    mapping(Tier => string) private _tierURIs;
+/// @title LingoNFT: An NFT Contract for Lingo SocialFi Campaign
+/// @notice This contract allows for the minting and management of various NFT tiers.
+/// @dev Inherits ERC721 for NFT functionality, Ownable for access control, and ReentrancyGuard for security.
+contract LingoNFT is ERC721, EIP712, Ownable {
+    using ECDSA for bytes32;
```
Comment (au2001) [inline contracts/LingoNFT.sol]: Unused?

File: scripts/Signer.js [hunk 13]
```diff
@@ -1,63 +1,67 @@
...
-const ECONOMY_SIGNER = "0x59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d";
-const BUSINESS_SIGNER = "0x5de4111afa1a4b94908f83103eb1f1706367c2e68ca870fc3fb9a804cdab365a";
-const FIRST_SIGNER = "0x689af8efa8c651a91ad287602527f3af2fe9f6501a7ac4b061667b5a93e037fd"   
-const BACKEND_VERIFY_URL = 'http://backend-url/api'; 
+const ECONOMY_SIGNER =
+  "0x59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d";
+const BUSINESS_SIGNER =
+  "0x5de4111afa1a4b94908f83103eb1f1706367c2e68ca870fc3fb9a804cdab365a";
+const FIRST_SIGNER =
+  "0x689af8efa8c651a91ad287602527f3af2fe9f6501a7ac4b061667b5a93e037fd";
```
Comment (au2001) [inline scripts/Signer.js]: Hardcoding private keys, very nice
I hope they're the ones in production

**Review by au2001 — CHANGES_REQUESTED** [review]
Better!
Please run Prettier though
You're probably missing the Solidity plugin
And there are a bunch of gas optimizations but not very important

File: contracts/LingoNFT.sol (hunk 11 above)
Comment (wepee) [reply contracts/LingoNFT.sol:145]: This is not a bug but a feature

File: scripts/Signer.js (hunk 13 above)
Comment (wepee) [reply scripts/Signer.js]: Haha we use KMS in production

File: contracts/LingoNFT.sol (hunk 12 above)
Comment (wepee) [reply contracts/LingoNFT.sol:22]: `ECDSA.recover(messageHash, v, r, s) != mintSigner`

File: contracts/LingoNFT.sol (hunk 11 above)
Comment (au2001) [reply contracts/LingoNFT.sol:145]: Ok just wanted to be sure
